src/convert-workspace.ts

```diff
@@ -91,48 +89,68 @@ function configureAnalyzer(options: WorkspaceConversionSettings) {
 }
 
 /**
- * Convert a set of workspace repos to JavaScript modules & npm.
+ * The results of a conversion, as a map of converted package npm names -> their
+ * full file path location on disk.
+ */
+export type ConversionResultsMap = Map<string, string>;
+
+/**
+ * Convert a set of workspace repos to npm packages and JavaScript modules.
+ * Returns a map of all packages converted, keyed by npm package name.
  */
-export default async function convert(options: WorkspaceConversionSettings) {
+export default async function convert(options: WorkspaceConversionSettings):
+    Promise<ConversionResultsMap> {
   const analyzer = configureAnalyzer(options);
   const analysis = await analyzer.analyzePackage();
   const htmlDocuments = [...analysis.getFeatures({kind: 'html-document'})];
   const conversionSettings = createDefaultConversionSettings(analysis, options);
   const urlHandler = new WorkspaceUrlHandler(options.workspaceDir);
   const converter = new ProjectConverter(urlHandler, conversionSettings);
+  const convertedPackageResults: ConversionResultsMap = new Map();
 
   // For each repo, convert the relevant HTML documents:
   for (const repo of options.reposToConvert) {
-    const repoDirName = repo.dir.split('/').pop()!;
-    const repoDocuments = htmlDocuments.filter((d) => {
-      return d.url.startsWith(repoDirName) &&
-          !conversionSettings.excludes.has(d.url);
-    });
-    for (const document of repoDocuments) {
+    const repoDirName = path.basename(repo.dir);
+    const bowerPath = path.join(repo.dir, 'bower.json');
+    const packageName = lookupNpmPackageName(bowerPath);
+    if (!packageName) {
+      continue;
+    }
+    for (const document of htmlDocuments) {
+      if (!document.url.startsWith(repoDirName) ||
```

Comment:
You shouldn't need this if you add `externalPackages: false` to the `getFeatures` call.